packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts

```diff
@@ -183,6 +189,198 @@ export const getAllLatestEntitiesResolver: ResolverFn<
   return entitySubgraph as Subgraph;
 };
 
+export const aggregateEntitiesResolver: Extract<
+  QueryResolvers<LoggedInGraphQLContext>["aggregateEntities"],
+  Function
+> = async (
+  _,
+  {
+    operation,
+    constrainsValuesOn,
+    constrainsPropertiesOn,
+    constrainsLinksOn,
+    constrainsLinkDestinationsOn,
+    isOfType,
+    hasLeftEntity,
+    hasRightEntity,
+  },
+  { dataSources },
+  __,
+) => {
+  const { graphApi } = dataSources;
+
+  /** @todo - should we throw an error, or should we just ignore the constraint and hope for the best */
```

Comment:
@CiaranMn any idea how we want to handle this at the moment? (regarding not having support for pagination)

Reply:
> Could we not just paginate the roots (skip and limit based on the pagination arguments) when returning them? At least then FE clients can be constructed to make calls for pagination and get progressive pages of the roots.

I guess we could.. I'm not sure if it would be a good thing. I'll try and think on this a little more.

> I know it will mean there is additional stuff in the subgraph but I don't think that matters – I may have forgotten an explanation of why it does in fact matter though.

This is a harder conversation, and one I was meaning to bring up in another forum today. @TimDiekmann and I were thinking about it in the [context of resolving the graph for various time intervals](https://github.com/hashintel/hash/pull/1848#discussion_r1072504403). I'll initiate the conversation somewhere with some context and tag you in it.

Reply:
I added an [Asana task](https://app.asana.com/0/1202805690238892/1203774687353264/f) _(internal)_ to track if we consider it as a bug if the subgraph contains more vertices/edges than requested.

Reply:
This is now redundant as pagination was removed